lib/nanoc/base/entities/content.rb

```diff
@@ -15,6 +15,10 @@ class Content
 
       # @param [String, nil] filename
       def initialize(filename)
+        if filename && !filename.start_with?('/')
+          raise ArgumentError, 'Content filename is not absolute'
+        end
```

Comment:
I can verify this does not work on windows; it throws for the filename "D:/site/content/htaccess.noext".
I was however able to get this to work using `!Pathname.new(filename).absolute?` instead of `!filename.start_with?('/')`.